### PR TITLE
fix: Removes the use of private headers from PDAL

### PR DESCRIPTION
**Pull request by MJohnson459**
While RenderDEM compiles when we are building from source, PDAL in conda does not include the private headers and so it fails to compile when using the conda version.

By switching to PDAL's public API, we unlock the ability to use a pre-compiled PDAL.

On the implementation side, I've tried to keep the changes as minimal as I can. We use the [writers.gdal stage](https://pdal.io/en/2.9.0/stages/writers.gdal.html#writers-gdal) which internally uses the same GDALGrid we were using. There is some added overhead in shaping the data for PDAL but from what I measured it isn't much.

[Writing with PDAL](https://pdal.io/en/2.9.0/development/writing.html)

To get a rough estimate of the overhead, I ran one of the lines I saw ODM run and used hyperfine to measure the performance but I will say I am still unsure exactly what the arguments do or if this is a feasible benchmark.

```bash
hyperfine renderdem ~/datasets/odm_data_aukerman/odm_georeferencing/odm_georeferenced_model.laz --outdir ~/datasets/odm_data_aukerman/odm_dem --output-type max --radiuses 0.04,0.05656854249492381,0.08000000000000002 --resolution 0.05 --max-tiles 0 --decimation 1 --classification -1 --tile-size 4096 --force
```

|      Implementation   | Time (mean ± σ) | User CPU |
|-------------------------------|-------------------------|--------------|
| Old: source PDAL, private API | 4.719s ± 0.124s | 18.974s  |
| New: conda PDAL, public API   | 5.127s ± 0.090s | 21.286s  |

Finally, the output tif looked reasonable from what I could tell. It's too large to attach here but I can upload it somewhere or share it some other way if needed.

### CODE REVIEW

File: render.cpp [hunk 1]
```diff
@@ -1,9 +1,14 @@
 #include <filesystem>
 #include <cmath>
 #include <algorithm>
-#include <array>
-#include "pdal/io/private/GDALGrid.hpp"
-#include "pdal/private/gdal/Raster.hpp"
```
Comment (DodgySpaniard) [inline render.cpp:6]: My concern here is that this is an obvious antipattern in the first place, so I wonder they did it in first place? Are there some serious trade off we may be missing?

**Review by DodgySpaniard — APPROVED** [review]
I do not see anything obviously wrong. There are a few things worth mentioning (but not blocking the diff). Basically:
 - I am not sure about behaviour parity, e.g. empty cells or bounds. It seems we are also relying on different pdal inner workings (which I am not familiar).
- The part were we do not have automated CI tests or a clear define process to flag regressions (either missed edge cases, final quality or compute resources) is a rather worrisome, albeit unrelated to this PR.

File: render.cpp (hunk 1 above)
Comment (smathermather) [reply render.cpp:6]: If I had to guess, 10% performance gain is probably the trade-off. But as this is a very small portion of the overall processing time on even larger projects, that penalty to correct a bad pattern is not hard to justify.

**Comment by smathermather** [comment]
> * I am not sure about behaviour parity, e.g. empty cells or bounds. It seems we are also relying on different pdal inner workings (which I am not familiar).

We should probably at least be checking for no points, though that enhancement is non-blocking for for this pull request.

**Review by smathermather — APPROVED** [review]

File: render.cpp (hunk 1 above)
Comment (MJohnson459) [reply render.cpp:6]: Right, performance is the most likely. I also considered if it was just due to the pattern of the library. PDAL is set up to be a "pipeline" of image functions but we only want this one operation. The two ways to access it were either private headers or paying the cost (perf and code) to set up the full pipeline.
